cpp/inc/bond/core/config.h

```diff
@@ -41,8 +41,23 @@
 #define BOND_CONSTEXPR_OR_CONST BOOST_CONSTEXPR_OR_CONST
 #define BOND_STATIC_CONSTEXPR   BOOST_STATIC_CONSTEXPR
 
-#ifdef BOND_HEADER_ONLY
+
+#define BOND_LIB_TYPE_HEADER    1
+#define BOND_LIB_TYPE_STATIC    2
+#define BOND_LIB_TYPE_DYNAMIC   3   // Not implemented
+
+#ifndef BOND_LIB_TYPE
+#define BOND_LIB_TYPE BOND_LIB_TYPE_HEADER
+#elif (BOND_LIB_TYPE != BOND_LIB_TYPE_HEADER) && (BOND_LIB_TYPE != BOND_LIB_TYPE_STATIC)
+#error Unknown library type is defined for BOND_LIB_TYPE
```

Comment:
"Unsupported" instead of "unknown"? `BOND_LIB_TYPE_DYNAMIC` isn't yet supported, but it isn't unknown.

Reply:
What if user really provides unknown one?

Reply:
An unknown one is also unsupported. ;-)

Reply:
OK.